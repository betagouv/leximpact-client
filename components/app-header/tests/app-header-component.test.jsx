import { createShallow } from "@material-ui/core/test-utils";

import AppHeaderComponent from "../app-header-component";

const CURRENT_NODE_ENV = process.env;

describe("components | app-header | index", () => {
  let shallow;
  beforeAll(() => {
    jest.resetModules();
    process.env = { ...CURRENT_NODE_ENV };
    delete process.env.NODE_ENV;

    const options = { dive: true };
    shallow = createShallow(options);
  });

  afterAll(() => {
    process.env = CURRENT_NODE_ENV;
  });

  describe("snapshot", () => {
    it("doit correspondre quand l'user n'est pas connecté", () => {
      const props = { isUserConnected: false };
      const wrapper = shallow(<AppHeaderComponent {...props} />);
      expect(wrapper).toBeDefined();
      expect(wrapper).toMatchSnapshot();
    });

    it("doit correspondre quand l'user est connecté", () => {
      const props = { isUserConnected: true };
      const wrapper = shallow(<AppHeaderComponent {...props} />);
      expect(wrapper).toBeDefined();
      expect(wrapper).toMatchSnapshot();
    });
  });
});
