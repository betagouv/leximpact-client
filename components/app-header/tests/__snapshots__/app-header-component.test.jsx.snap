// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components | app-header | index snapshot doit correspondre quand l'user est connecté en mode desktop 1`] = `
<WithStyles(ForwardRef(AppBar))
  position="relative"
>
  <WithStyles(ForwardRef(Toolbar))
    classes={
      Object {
        "root": "AppHeader-toolbarRoot-7",
      }
    }
  >
    <WithStyles(HeaderMenuButton) />
    <WithStyles(ForwardRef(Typography))
      classes={
        Object {
          "root": "AppHeader-titleRoot-6",
        }
      }
      component="div"
    >
      <span>
        <span
          className="AppHeader-lighterMobileTitle-3"
        >
          LEXIMPACT 
        </span>
      </span>
    </WithStyles(ForwardRef(Typography))>
    <div />
  </WithStyles(ForwardRef(Toolbar))>
</WithStyles(ForwardRef(AppBar))>
`;

exports[`components | app-header | index snapshot doit correspondre quand l'user est connecté en mode mobile 1`] = `
<WithStyles(ForwardRef(AppBar))
  position="relative"
>
  <WithStyles(ForwardRef(Toolbar))
    classes={
      Object {
        "root": "AppHeader-toolbarRoot-7",
      }
    }
  >
    <WithStyles(HeaderMenuButton) />
    <WithStyles(ForwardRef(Typography))
      classes={
        Object {
          "root": "AppHeader-titleRoot-6",
        }
      }
      component="div"
    >
      <span>
        <span
          className="AppHeader-lighterMobileTitle-3"
        >
          LEXIMPACT 
        </span>
      </span>
    </WithStyles(ForwardRef(Typography))>
    <div />
  </WithStyles(ForwardRef(Toolbar))>
</WithStyles(ForwardRef(AppBar))>
`;

exports[`components | app-header | index snapshot doit correspondre quand l'user n'est pas connecté en mode desktop 1`] = `
<WithStyles(ForwardRef(AppBar))
  position="relative"
>
  <WithStyles(ForwardRef(Toolbar))
    classes={
      Object {
        "root": "AppHeader-toolbarRoot-7",
      }
    }
  >
    <WithStyles(HeaderMenuButton) />
    <WithStyles(ForwardRef(Typography))
      classes={
        Object {
          "root": "AppHeader-titleRoot-6",
        }
      }
      component="div"
    >
      <div>
        <div
          className="AppHeader-lighterTitle-4"
        >
          LEXIMPACT 
        </div>
      </div>
    </WithStyles(ForwardRef(Typography))>
    <div />
  </WithStyles(ForwardRef(Toolbar))>
</WithStyles(ForwardRef(AppBar))>
`;

exports[`components | app-header | index snapshot doit correspondre quand l'user n'est pas connecté en mode mobile 1`] = `
<WithStyles(ForwardRef(AppBar))
  position="relative"
>
  <WithStyles(ForwardRef(Toolbar))
    classes={
      Object {
        "root": "AppHeader-toolbarRoot-7",
      }
    }
  >
    <WithStyles(HeaderMenuButton) />
    <WithStyles(ForwardRef(Typography))
      classes={
        Object {
          "root": "AppHeader-titleRoot-6",
        }
      }
      component="div"
    >
      <span>
        <span
          className="AppHeader-lighterMobileTitle-3"
        >
          LEXIMPACT 
        </span>
      </span>
    </WithStyles(ForwardRef(Typography))>
    <div />
  </WithStyles(ForwardRef(Toolbar))>
</WithStyles(ForwardRef(AppBar))>
`;
